libnetwork/sandbox_dns_unix.go

```diff
@@ -134,10 +136,11 @@ func (sb *Sandbox) buildHostsFile() error {
 		return nil
 	}
 
-	extraContent := make([]etchosts.Record, 0, len(sb.config.extraHosts))
+	extraContent := make([]etchosts.Record, 0, len(sb.config.extraHosts)+len(ifaceIPs))
 	for _, extraHost := range sb.config.extraHosts {
 		extraContent = append(extraContent, etchosts.Record{Hosts: extraHost.name, IP: extraHost.IP})
 	}
+	extraContent = append(extraContent, sb.makeHostsRecs(ifaceIPs)...)
 
 	// Assume IPv6 support, unless it's definitely disabled.
 	buildf := etchosts.Build
```

Comment:
> ☝️ was also curious (because it mentions "definitely disabled"), should it also skip if !ok, or only if !en ?

If `!ok`, maybe there is IPv6 in the container, but it can't be sure. So, it includes the IPv6 records by calling `Build` instead of `BuildNoIPv6` ... I think it's alright.